backend/app/tests/api/routes/schedule/test_post_schedule_api.py

```diff
@@ -0,0 +1,155 @@
+import re
+
+from fastapi.testclient import TestClient
+
+from app.core.config import settings
+
+first_valid_schedule = {
+    "schedule": {
+        "startDate": "2024-10-02",
+        "endDate": "2024-11-01",
+        "daysBetween": 1,
+        "skipWeekends": "false",
+        "skipHolidays": "false",
+        "timesOfDay": ["08:00"],
+    }
+}
+
+second_valid_schedule = {
+    "schedule": {
+        "startDate": "2024-12-12",
+        "endDate": "2024-12-22",
+        "daysBetween": 1,
+        "skipWeekends": "true",
+        "skipHolidays": "true",
+        "timesOfDay": ["18:00"],
+    }
+}
+
+schedule_with_missing_attribute = {
+    "schedule": {
+        "startDate": "2024-12-12",
+        "endDate": "2024-12-22",
+        "daysBetween": 1,
+        "skipWeekends": "true",
+        "skipHolidays": "false",
+        "timesOfDay": ["six o'clock"],
+    }
+}
+
+schedule_with_bad_time = {
+    "schedule": {
+        "startDate": "2024-12-12",
+        "endDate": "2024-12-22",
+        "daysBetween": 1,
+        "skipWeekends": "true",
+        "skipHolidays": "false",
+        "timesOfDay": ["six o'clock"],
+    }
+}
+
+schedule_with_bad_date = {
+    "schedule": {
+        "startDate": "December 10 2024",
+        "endDate": "2024-12-22",
+        "daysBetween": 1,
+        "skipWeekends": "true",
+        "skipHolidays": "false",
+        "timesOfDay": ["18:00"],
+    }
+}
+
+not_a_schedule_string = "why do birds suddenly appear every time you are near?"
+
+
+def test_create_schedule_when_there_is_no_schedule_should_make_new_schedule(
+    client: TestClient, superuser_token_headers: dict[str, str]
+) -> None:
+    response = client.post(
```

Comment:
Can this duplicated code be refactored into a helper?

Reply:
Will do!

Reply:
Maybe you can copy paste from `api_post` function here? https://github.com/Incredihire/Survey/pull/90/files#diff-0da1dcaf359d94384308badf89a6a7f2fff7cf697a3dd7047f88bc49f9d5e185

Reply:
I created my own utils file.  Thanks though!